### PR TITLE
Update Docker image builds to use newer tooling

### PR DESCRIPTION
**Pull request by damonbarry**
Docker recently introduced [provenance attestation](https://docs.docker.com/build/attestations/slsa-provenance/) in buildx 0.10.0, which broke our multi-arch image builds. To build our multi-arch manifests, we use a really old tool that isn't compatible with the buildx changes. To work around the problem, we disabled provenance attestation in our builds (see f82a7d94f37a795ce908c67dca8192e626b38788).

With this change, we re-enable provenance attestation and upgrade our multi-arch build tooling to support it.

I also took this opportunity to refactor our Dockerfiles into a single Dockerfile per module except API Proxy (which structures its x64 and arm images very differently). I initially did this refactoring because I planned to upgrade our pipelines/scripts to the canonical method for building Docker images: passing a comma-separated list of platforms to `docker buildx build` and letting it build the images _and_ the multi-arch manifest at once. This method would have required a single Dockerfile for all architectures. However, since we provide per-architecture tags for our images, it ended up being easier to keep our current pattern (build each single-arch image, then create the multi-arch manifest separately) and just upgrade the tools.

Other changes:
- In cases where I was already updating a pipeline script task (e.g., to add/change a parameter passed to a script), I converted the task syntax to use the newer 'script' alias for consistency with other pipelines.
- In our API Proxy pipelines, there were three different places where we installed qemu and binfmt to prepare for cross-compiling the code, but the agent already has those installed so I removed it.
- Also in our API Proxy pipelines, there was a place where we did some docker buildx setup even though the job doesn't use docker buildx. Maybe it used to? Anyway, I removed the setup task.
- Removed the explicit bin_dir argument from all image-linux.yaml calls, since it is redundant with that parameter's default value.
- Deleted all manifest.yaml.template files, since they were required for the old manifest tool and are no longer used.
- In some of our release pipelines we have a default value for the 'tags' parameter that is given in not-quite-JSON format (an array with a single string value in _single_ quotes). I changed it to valid JSON so I can use jq to merge it with other tags when building the multi-arch image.
- Removed the unused '--postfix' parameter from buildRocksDb.sh.
- Removed the redundant and misnamed '--image-name' parameter from buildApiProxy.sh.
- Cleaned up script variables and args parsing for consistency, and in a few cases to fix minor bugs.

To test, I ran the CI build, which exercises the key YAML templates and all the scripts. I also ran the end-to-end tests and nested end-to-end tests to verify that the images still work as expected.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [X] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [X] Title of the pull request is clear and informative.
- [X] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [X] concise summary of tests added/modified
	- [X] local testing done.

### CODE REVIEW

**Review by and-rewsmith — APPROVED** [review]

File: doc/devguide.md [hunk 1]
```diff
@@ -105,20 +105,8 @@ Sometimes it is useful to build the Edge Hub container locally. If you want to d
 ## Build Manifest Image
 There is a script in the repo to build multi-architecture images.
 This script assumes that the platform specific images are already in the docker registry.
-Usage is as follows:
-```sh
-$ scripts/linux/buildManifest.sh --help
-
-buildManifest.sh [options]
-Note: Depending on the options you might have to run this as root or sudo.
+Call `scripts/linux/buildManifest.sh --help` for more information.
```
Comment (and-rewsmith) [inline doc/devguide.md:108]: Does the buildLocalEdgeHub.sh still work (referenced above)?

File: scripts/linux/buildManifest.sh [hunk 2]
```diff
@@ -117,26 +143,26 @@ process_args()
...
+IFS=',' read -a architectures <<< "$ARCH"
+for arch in ${architectures[@]}
+do
+    image="$image_name:$DOCKER_IMAGEVERSION-linux-$(convert_arch $arch)"
+    arch_digests+=( $(docker buildx imagetools inspect $image --format '{{json .Manifest}}' |
+        jq --arg arch "$arch" -r '($arch | split("/")) as $parts |
+            .manifests[] |
+            select(.platform.architecture == $parts[0]) |
+            if ($parts | length > 1) then select(.platform.variant == $parts[1]) else . end |
+            .digest') )
```
Comment (onalante-msft) [inline scripts/linux/buildManifest.sh:158]:
```suggestion
        jq --arg arch "$arch" -r '.manifests[] |
            select($arch == ([.platform | (.architecture, .variant // empty)] | join("/"))) |
            .digest') )
```

**Review by onalante-msft — DISMISSED** [review]

File: doc/devguide.md (hunk 1 above)
Comment (damonbarry) [reply doc/devguide.md:108]: Honestly, I didn't try it. 😞 I'll give it a shot.

File: scripts/linux/buildManifest.sh (hunk 2 above)
Comment (damonbarry) [reply scripts/linux/buildManifest.sh]: Love it. Thanks!

**Review by onalante-msft — APPROVED** [review]

File: doc/devguide.md (hunk 1 above)
Comment (damonbarry) [reply doc/devguide.md:108]: @and-rewsmith I just confirmed that `scripts/linux/buildLocalEdgeHub.sh` works as expected. ✅

**Pull request merged**